### PR TITLE
[1] Initial Commit of OpenAPI Tools for Eclipse

### PR DESCRIPTION
**Pull request by keithchong**
[1] Initial Commit of OpenAPI Tools for Eclipse
Signed-off-by: Keith Chong <kchong@ca.ibm.com>

### CODE REVIEW

**Comment by keithchong** [comment]
Yes, the two jars are removed

File: dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/Constants.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
```
Comment (sghung) [inline dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/Constants.java]: Can we use something like  @SuppressWarning("nls") to handle all the constants here?

File: dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/commands/AbstractOpenApiGeneratorCommand.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+			pluginLocation = coreURL.getFile();
+			if (isWindows && pluginLocation != null && pluginLocation.startsWith("/")) { //$NON-NLS-1$
+				pluginLocation = pluginLocation.substring(1);
+			}
+		} catch (IOException e) {
+			return new Status(IStatus.ERROR, org.eclipse.codewind.openapi.ui.Activator.PLUGIN_ID, e.getMessage());
+		}
+		
+		String jarOverride = System.getProperty(IOpenApiConstants.OPENAPI_CLI_OVERRIDE_PROPERTY);
+		// eg. -Dorg.ecliipse.codewind.openapi.cli.jar.path=/Users/kchong/openapi/openapi-generator-cli-4.0.0.jar
```
Comment (sghung) [inline dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/commands/AbstractOpenApiGeneratorCommand.java]: Just confirming, do you want your OS username put on this comment Keith?

File: dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/wizard/AbstractGenerateWizardPage.java [hunk 3]
```diff
@@ -0,0 +1,400 @@
+/*******************************************************************************
```
Comment (sghung) [inline dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/wizard/AbstractGenerateWizardPage.java]: Missing some non NLS comments here

File: dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/commands/AbstractOpenApiGeneratorCommand.java (hunk 2 above)
Comment (keithchong) [reply dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/commands/AbstractOpenApiGeneratorCommand.java]: It was intentional and served as a real-world example.  I could change it to /Users/foo/

**Comment by keithchong** [comment]
@sghung , all comments addressed except for the @SuppressWarnings("nls") on Constants.java.  We're not translating this so I think we want to enable the warnings only when we need to translate.   I just simply put a comment in that class to indicate the strings are not translatable for now.

**Comment by keithchong** [comment]
Elson/Steven, I believe we are OK to push the changes.

**Comment by keithchong** [comment]
Hi @elsony , the jars have been removed and Steven approved already.  It's ready to be merged.

"Only those with write access to this repository can merge pull requests."

**Pull request merged**